docs/OpenTypeFeatureFileSpecification.md

```diff
@@ -825,14 +826,37 @@ DEU
 
 Note that the final space in the example is implicit.
 
+A tag can only have characters from the following set:
+
+    ABCDEFGHIJKLMNOPQRSTUVWXYZ
+    abcdefghijklmnopqrstuvwxyz
+    0123456789
+    .  # period
+    _  # underscore
+    !  # Exclamation point
+    $  # Dollar sign
+    %  # Percent sign
+    &  # Ampersand
+    *  # Asterisk
+    +  # Plus sign
+    :  # Colon
+    ?  # Question mark
+    ^  # Caret
+    '  # Back-quote
+    |  # Vertical bar
+    ~  # Tilde
+
+and must not start with a digit or hyphen. However, use of characters beyond
+those in production glyph names is not recommended.
```

Comment:
OpenType tags have [different requirements](https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types) than what is stated here:
> A Tag value is a uint8 array. Each byte within the array must have a value in the range 0x20 to 0x7E. This corresponds to the range of values of Unicode Basic Latin characters in UTF-8 encoding, which is the same as the printable ASCII characters. As a result, a Tag value can be re-interpreted as a four-character sequence, which is conventionally how they are referred to. Formally, however, the value is a byte array. When re-interpreted as characters, the Tag value is case sensitive. It must have one to four non-space characters, padded with trailing spaces (byte value 0x20). A space character cannot be followed by a non-space character.

Reply:
Technically tags with that degree of generality weren't supported by the previous implementation -- it skipped the semicolon.

Supporting *unescaped* tags with that full range would either require going back to a mixed parser-lexer (an anti-goal of this port) or would greatly complicate the grammar. It was felt that the more limited set would be fine in practice.

An alternative, which would be straightforward to add, would be to also allow strings in the tag position for tags beyond those in the more limited character set (with escaped quotes if necessary).

Reply:
At least the new implementation should support the same tags supported by the previous implementation, otherwise it would be a breaking change and would limit makeotf from accepting otherwise valid OpenType tags (it might not be an real issue now, but who knows when such tags would be required in the future).

Reply:
@khaledhosny regarding breaking changes: we recognize this, and for that reason have opted to bump the version number of the spec and have added appropriate documentation. An implementation that prefers the older scheme is free to peg their level of support to an older version of the spec (or use an older version of AFDKO/makeotfexe that implements the older spec).

Reply:
Fair enough.

Reply:
It does seem appropriate for us to note these changes prominently once we get this merged and into the release process.